[BUG] index out of range [0] with length 0 at modules/setting

### Description

The function `getPostgreSQLConnectionString` in modules/setting/database.go parse the dbhost, and it works like:
```
host, port := parsePostgreSQLHostPort(dbHost)
if host[0] == '/' { // looks like a unix socket
...
```
but it just split host-string with `:`. If the provided host-string starts with `:`, the host will be "", and it will trigger an error like `index out of range [0] with length 0` 

The poc is here:
```go
package main

import "code.gitea.io/gitea/modules/setting"

func main() {
	setting.Database.Type = "postgres"
	setting.Database.Host = ":00000"
	_, _ = setting.DBConnStr()
}

```

And it crashed as :
```go
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
code.gitea.io/gitea/modules/setting.getPostgreSQLConnectionString({0x744b0d, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x743130, 0x1}, ...)
        /home/zjx/workspace/gowork/src/go-fdg-exmaples/gitea/modules/setting/database.go:158 +0x410
code.gitea.io/gitea/modules/setting.DBConnStr()
        /home/zjx/workspace/gowork/src/go-fdg-exmaples/gitea/modules/setting/database.go:122 +0x5e9
main.main()
        /home/zjx/workspace/gowork/src/go-fdg-exmaples/gitea/modules/setting/pocTest_getPostgreSQL/main.go:8 +0x8a
exit status 2
```

### Gitea Version

with git commit as ee769f7459ffba310563cd735f503d3d7576da60

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

I test the API in Gitea through go-fuzz

### Database

_No response_

--- a/modules/setting/database.go
+++ b/modules/setting/database.go
@@ -179,6 +179,7 @@ func getPostgreSQLConnectionString(dbHost, dbUser, dbPasswd, dbName, dbParam, db
 
 // ParseMSSQLHostPort splits the host into host and port
 func ParseMSSQLHostPort(info string) (string, string) {
+	// the default port "0" might be related to MSSQL's dynamic port, maybe it should be double-confirmed in the future
 	host, port := "127.0.0.1", "0"
 	if strings.Contains(info, ":") {
 		host = strings.Split(info, ":")[0]

--- a/modules/setting/database.go
+++ b/modules/setting/database.go
@@ -156,6 +156,12 @@ func parsePostgreSQLHostPort(info string) (string, string) {
 	} else if len(info) > 0 {
 		host = info
 	}
+	if host == "" {
+		host = "127.0.0.1"
+	}
+	if port == "" {
+		port = "5432"
+	}
 	return host, port
 }
 
@@ -182,6 +188,12 @@ func ParseMSSQLHostPort(info string) (string, string) {
 		port = strings.TrimSpace(strings.Split(info, ",")[1])
 	} else if len(info) > 0 {
 		host = info
+	}
+	if host == "" {
+		host = "127.0.0.1"
+	}
+	if port == "" {
+		port = "0"
 	}
 	return host, port
 }

